### PR TITLE
Support destroying slices

### PR DESCRIPTION
**Pull request by Nesurion**
Add support for supplying sliceIDs when using the destroy command. See #149 for more information.

### CODE REVIEW

**Comment by Nesurion** [comment]
@giantswarm/cena RFR

File: cli/destroy.go [hunk 1]
```diff
@@ -20,24 +20,29 @@ var (
...
 		os.Exit(1)
 	}
 
+	var err error
 	newRequestConfig := controller.DefaultRequestConfig()
-	newRequestConfig.Group = group
-	req := controller.NewRequest(newRequestConfig)
-
-	req, err := newController.ExtendWithExistingSliceIDs(req)
+	newRequestConfig.Group, newRequestConfig.SliceIDs, err = parseGroupCLIargs(args)
```
Comment (zeisss) [inline cli/destroy.go]: parseGroupCLIArgs

File: cli/destroy.go [hunk 2]
```diff
@@ -10,7 +10,7 @@ import (
 
 var (
 	destroyCmd = &cobra.Command{
-		Use:   "destroy <group|slice...>",
+		Use:   "destroy <group[@slice]...>",
```
Comment (JosephSalisbury) [inline cli/destroy.go:13]: What does the "@" mean?

File: cli/request_test.go [hunk 3]
```diff
@@ -151,3 +152,71 @@ func Test_Request_ExtendWithContent(t *testing.T) {
...
+
+func Test_Request_ParseGroupCLIargs(t *testing.T) {
+	type Expected struct {
+		Group    string
+		SliceIDs []string
+	}
+	testCases := []struct {
+		Input    []string
+		Expected Expected
+		Errorf   func(error) bool
```
Comment (zeisss) [inline cli/request_test.go]: Errorf? Looks like Printf.

File: cli/request_test.go [hunk 4]
```diff
@@ -151,3 +152,71 @@ func Test_Request_ExtendWithContent(t *testing.T) {
...
+		// Tests that multiple group sliceIDs are split correctly
+		{
+			Input: []string{"mygroup@1", "mygroup@2"},
+			Expected: Expected{
+				Group:    "mygroup",
+				SliceIDs: []string{"1", "2"},
+			},
+			Errorf: nil,
+		},
+		// Tests that mixed groups return an invalidArgumentsError
```
Comment (zeisss) [inline cli/request_test.go]: How about we split this into two test functions? One for success cases, one for failure cases? We don't need Expected for the latter, nor Errorf for the first.

File: cli/request_test.go [hunk 5]
```diff
@@ -151,3 +152,71 @@ func Test_Request_ExtendWithContent(t *testing.T) {
...
+	type Expected struct {
+		Group    string
+		SliceIDs []string
+	}
+	testCases := []struct {
+		Input    []string
+		Expected Expected
+		Errorf   func(error) bool
+	}{
+		// Tests that on sliceIDs are returned when non where provided
```
Comment (JosephSalisbury) [inline cli/request_test.go]: no sliceIDs?

Comment (JosephSalisbury) [inline cli/request_test.go]: `Tests that no sliceIDs are returned when none were provided`

File: cli/destroy.go (hunk 2 above)
Comment (Nesurion) [inline cli/destroy.go:13]: its the separator as discussed in https://github.com/giantswarm/inago/issues/149

File: cli/request_test.go (hunk 4 above)
Comment (JosephSalisbury) [inline cli/request_test.go]: What about mixed groups, with slice ids?

File: cli/request_test.go [hunk 6]
```diff
@@ -151,3 +152,77 @@ func Test_Request_ExtendWithContent(t *testing.T) {
...
+		if group != test.Expected.Group {
+			t.Fatalf("got group %v, expected group to be %v.", group, test.Expected.Group)
+		}
+		if !reflect.DeepEqual(sliceIDs, test.Expected.SliceIDs) {
+			t.Fatalf("got sliceIDs %v, expected sliceIDs to be %v.", sliceIDs, test.Expected.SliceIDs)
+		}
+	}
+}
+
+func Test_Request_ParseGroupCLIargs_Error(t *testing.T) {
```
Comment (zeisss) [inline cli/request_test.go]: Args

**Comment by zeisss** [comment]
LGTM

**Pull request merged**